codec-http2/src/main/java/io/netty/handler/codec/http2/HpackDecoder.java

```diff
@@ -119,24 +120,25 @@
      * This method assumes the entire header block is contained in {@code in}.
      */
     public void decode(int streamId, ByteBuf in, Http2Headers headers, boolean validateHeaders) throws Http2Exception {
-        Http2HeadersSink sink = new Http2HeadersSink(headers, maxHeaderListSize);
-        decode(in, sink, validateHeaders);
+        Http2HeadersSink sink = new Http2HeadersSink(streamId, headers, maxHeaderListSize, validateHeaders);
```

Comment:
I know it wasn't introduced by this PR but it would be nice to not have to introduce object allocation for each decode. Seems like this can be handled as local state in the now private `decode` method.

Reply:
Do you mind if we defer this for a later commit? I think the private `decode` method is really heavy at this time and don't really want to add more state to it, at least as part of another behavior change.

fwiw, I suspect this is a really good candidate for stack allocation and even if it's not, it's super short lived garbage that only happens once on a complete HEADERS block.

Reply:
I am +1 to defer this for a followup as its not introduced by this change